Public/Src/Engine/Scheduler/FileMonitoringViolationAnalyzer.cs

```diff
@@ -620,37 +650,134 @@ public ReportedViolation(bool isError, DependencyViolationType type, AbsolutePat
-                return (IsWriteViolation ? ReportedFileAccess.WriteDescriptionPrefix : ReportedFileAccess.ReadDescriptionPrefix) + Path.ToString(pathTable);
-            }
-
-            /// <summary>
-            /// Determines whether the current violation is a write violation
+            /// A simplified violation type for sake of displaying in the DisallowedFileAccess summary message
             /// </summary>
-            public bool IsWriteViolation
+            public SimplifiedViolationType ReportingType
             {
                 get
                 {
                     switch (Type)
                     {
                         case DependencyViolationType.DoubleWrite:
+                            return SimplifiedViolationType.DoubleWrite;
+                        case DependencyViolationType.ReadRace:
+                        case DependencyViolationType.UndeclaredOrderedRead:
+                        case DependencyViolationType.MissingSourceDependency:
+                        case DependencyViolationType.UndeclaredReadCycle:
+                        case DependencyViolationType.ReadUndeclaredOutput:
+                            return SimplifiedViolationType.Read;
                         case DependencyViolationType.UndeclaredOutput:
-                        case DependencyViolationType.WriteOnAbsentPathProbe:
+                        case DependencyViolationType.WriteInSourceSealDirectory:
                         case DependencyViolationType.WriteInUndeclaredSourceRead:
                         case DependencyViolationType.WriteInExistingFile:
-                            return true;
+                        case DependencyViolationType.WriteToTempPathInsideSharedOpaque:
+                        case DependencyViolationType.WriteOnAbsentPathProbe:
+                            return SimplifiedViolationType.Write;
+                        case DependencyViolationType.AbsentPathProbeUnderUndeclaredOpaque:
+                            return SimplifiedViolationType.Probe;
+                        default:
+                            throw new NotImplementedException("Need to implement for: " + Type.ToString());
+                    }   
+                }
+            }
+
+            /// <summary>
+            /// Renders the violation for display in the DFA summary
+            /// </summary>
+            public string RenderForDFASummary(PathTable pathTable)
+            {
+                string path = Path.ToString(pathTable);
+                switch (ReportingType)
+                {
+                    case SimplifiedViolationType.Probe:
+                        return " P  " + path;
+                    case SimplifiedViolationType.Read:
+                        return " R  " + path;
+                    case SimplifiedViolationType.Write:
+                        return " W  " + path;
+                    case SimplifiedViolationType.DoubleWrite:
+                        return " DW  " + path;

```

Comment:
Nit: DW has 5 spaces then the path, so it will be slightly offset when printed, maybe adjust the others too?

Reply:
Ooops. Meant to remove one of its trailing spaces to keep the paths lined up. Fixed.

---
In reply to: [316777924](https://github.com/microsoft/BuildXL/pull/787#discussion_r316777924) [](ancestors = 316777924)